app/controllers/activities_controller.rb

```diff
@@ -30,4 +30,38 @@ def poll
       format.js
     end
   end
+
+  private
+
+  def period_start
+    DateTime.parse params[:period_start]
+  end
+
+  def period_end
+    if params[:period_end].empty?
+      DateTime.now
+    else
+      DateTime.parse params[:period_end]
+    end
+  end
+
+  def filtering_params
+    params.slice(:user, :type, :period_start, :period_end)
+  end
+
+  def filter_activities(activities)
+    filtering_params.each do |key, value|
+      next if value.blank?
+
+      if key == 'period_start'
+        value = DateTime.parse(value).beginning_of_day
+      elsif key == 'period_end'
+        value = DateTime.parse(value).end_of_day
+      end
```

Comment:
I think we need to add an `else` here otherwise, if an evil user can pass any param here and it will still be called by `activities.send("by_#{key}", value)`

Reply:
Since we will be using Activity.since(date) & Activity.before(date) there is no risk here of sending dynamic methods anymore.